internal/auth/internal/types/traffic_permissions.go

```diff
@@ -17,7 +18,12 @@ func RegisterTrafficPermissions(r resource.Registry) {
 		Proto:    &pbauth.TrafficPermissions{},
 		Scope:    resource.ScopeNamespace,
 		Validate: ValidateTrafficPermissions,
-		Mutate:   MutateTrafficPermissions,
+		ACLs: &resource.ACLHooks{
+			Read:  aclReadHookTrafficPermissions,
+			Write: aclWriteHookTrafficPermissions,
+			List:  aclListHookTrafficPermissions,
+		},
+		Mutate: MutateTrafficPermissions,
```

Comment:
supernit: keep the mutate hook next to the validate hook since they are closely related.